### DISCUSSION TITLE
Zerocopy with sequence datatype

### DISCUSSION THREAD

**Discussion by ussefdesouky** [eProsima/Fast-DDS#3810] [Support]
### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

I am using the zerocopy example, just changed the message struct and added a `sequence<octet> img;`

### Current behavior

The Zerocopy example totally depending on the following line of code, so if it's true it publishes the message.
ReturnCode_t::RETCODE_OK == writer_->loan_sample(sample)

### Steps to reproduce

when I change the normal .idl message from
struct LoanableHelloWorld
{
	unsigned long index;
	unsigned long long timestamp;
	char data[256];
};
to the following
`struct LoanableHelloWorld
{
	unsigned long index;
	unsigned long long timestamp;
	sequence<octet> img;
};`

this line of code returns false so it does not publish any data again.

Is the sequence datatype can't be used in such transport protocol?

### Fast DDS version/commit

master

### Platform/Architecture

Other. Please specify in Additional context section.

### Transport layer

Zero copy

### Relevant log output

```shell
Starting 
LoanableHelloWorld DataWriter created.
LoanableHelloWorld DataWriter waiting for DataReaders.
DataWriter matched.
```

**Comment by JLBuenoLopez** [comment]
Thanks for the report @ussefdesouky 

Please, take into account the [zero-copy mechanism constraints](https://fast-dds.docs.eprosima.com/en/latest/fastdds/use_cases/zero_copy/zero_copy.html#constraints). The modification introduced in the IDL file prevents both the plain and bounded requirements to be met. Besides, the `sequence` keyword requires as a non-optional parameter the type of the sequence, so the given IDL is malformed.

I am moving this to the support discussion forum as per [Fast DDS CONTRIBUTING guidelines](https://github.com/eProsima/Fast-DDS/blob/master/CONTRIBUTING.md#issues-and-support).

**Reply by ussefdesouky** [reply]
I have updated the idl message content it was written in wrong form. Also, you mean that 
`struct LoanableHelloWorld
{
	sequence<octet> img;
};`
this form can work with no issues.

**Reply by JLBuenoLopez** [reply]
The sequence is not bounded, so no, zero-copy mechanism is not going to work with that sequence. Data-sharing mechanism requires for the type to be bounded. [Fast DDS provides API](https://fast-dds.docs.eprosima.com/en/latest/fastdds/api_reference/dds_pim/topic/typesupport.html#_CPPv4NK8eprosima7fastdds3dds11TypeSupport10is_boundedEv) to check if the type is bounded and plain. If those methods return true, then the type might be used with zero-copy mechanism.

**Reply by ussefdesouky** [reply]
Thanks a lot, quite clear now. If I am going to publish image frame with the least latency which transport protocol you would recommend?

Thanks agian, Bueno.
